src/net/java/sip/communicator/impl/protocol/jabber/extensions/colibri/ColibriBuilder.java

```diff
@@ -1149,103 +1212,69 @@ public void setSimulcastMode(SimulcastMode simulcastMode)
     }
 
     /**
-     * Returns the channel from {@link #request} from a particular content
-     * and with a particular ID. If the request does not contain such a channel,
-     * a new instance is created and added to {@link #request}, and it is
-     * initialized only with the fields required to identify the particular
-     * Colibri channel on the bridge (i.e. just the ID is specified).
+     * Returns the channel from {@link #request} which matches a particular
+     * content name and a particular {@link ColibriConferenceIQ.ChannelCommon}
+     * instance (i.e. their IDs match).
+     * If the request does not contain such a channel, a new instance is created
+     * and added to {@link #request}, and it is initialized only with the fields
+     * required to identify the particular Colibri channel on the bridge (i.e.
+     * just the ID is specified). The exact run-time type of the instance (
+     * {@link ColibriConferenceIQ.Channel},
+     * {@link ColibriConferenceIQ.SctpConnection}, or
+     * {@link ColibriConferenceIQ.OctoChannel}) depends on the type of
+     * {@code channel}.
      *
-     * The returned value is always a non-null instance of
-     * {@link ColibriConferenceIQ.Channel} if {@code sctpConnection} is
-     * {@code false} and a non-null instance of
-     * {@link ColibriConferenceIQ.SctpConnection} otherwise.
+     * The returned value is always non-null.
      *
      * @param contentName the name of the content.
-     * @param channelId the ID of the channel.
-     * @param sctpConnection whether to create a
-     * {@link ColibriConferenceIQ.SctpConnection} or a
-     * {@link ColibriConferenceIQ.Channel}, if a new channel is to be created.
+     * @param channel the channel to match.
      *
      * @return the channel from {@link #request} with the specified content
-     * name and ID.
+     * name and ID and run-time type matching this of {@code channel}.
      */
-    private ColibriConferenceIQ.ChannelCommon getRequestChannel(
-            String contentName, String channelId, boolean sctpConnection)
+    private <T extends ColibriConferenceIQ.ChannelCommon> T getRequestChannel(
+        String contentName, T channel)
```

Comment:
it's a bummer we have to pass the whole thing in instead of just the id.  i was thinking we could pass the id and a .class and use reflection to achieve this (it would also remove the need for the if/else statement), but maybe that's going to far to try and avoid passing the whole object.

Reply:
I mostly agree, but since it's a private method and we have the ChannelCommon instance everywhere where we need it, I thought that was clean enough